packages/vite/src/node/plugin.ts

```diff
@@ -121,7 +121,7 @@ export interface Plugin extends RollupPlugin {
     this: PluginContext,
     source: string,
     importer: string | undefined,
-    options: { custom?: CustomPluginOptions; ssr?: boolean }
+    options: { custom?: CustomPluginOptions; ssr?: boolean; scan?: boolean }
```

Comment:
Are we fine with adding a new `scan` option here? This looks to be a public API. Perhaps we should document it or mark it as internal?

I think this is great as a basis for a more flexible scanner powered by Vite plugins in the future, and could expand to pre-bundling too, but it might be too early to expose the API before we can define its usecase? 🤔

Reply:
Let's mark it as `@internal` so we can merge this without going through a team meeting. We can create later a PR to remove the internal status and document it. I also think it is good that plugins know if they are being called during scan or during normal flow.

About naming, should it be `scan`, `scanning`, or something else?